zkml/src/layers/transformer/logits.rs

```diff
@@ -0,0 +1,82 @@
+use crate::argmax_slice;
+use anyhow::ensure;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    Tensor,
+    layers::provable::{Evaluate, LayerOut, OpInfo},
+    tensor::Number,
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum Logits {
+    Argmax,
+}
+
+impl<N: Number> Evaluate<N> for Logits {
+    fn evaluate<E: ff_ext::ExtensionField>(
+        &self,
+        inputs: &[&crate::Tensor<N>],
+        _unpadded_input_shapes: Vec<Vec<usize>>,
+    ) -> anyhow::Result<crate::layers::provable::LayerOut<N, E>> {
+        match self {
+            Logits::Argmax => {
+                let indices = inputs
+                    .iter()
+                    .map(|input| {
+                        ensure!(
+                            input.get_shape().len() >= 2,
+                            "Argmax is for tensors of rank >= 2"
+                        );
+                        let last_row = input.slice_on_dim(1).last().unwrap();
```

Comment:
Why taking only the last row? Is it because we are assuming that all tokens except the last one are input tokens or already processed tokens?

Reply:
Well, for inference, we only care about the last one. But i agree the API could take everything...

Reply:
Changed it to running argmax on all the last dimension slices of all the input tensors.